prov/sockets/src/sock_progress.c

```diff
@@ -1325,7 +1341,7 @@ static int sock_pe_process_rx_send(struct sock_pe *pe,
 	data_len = pe_entry->msg_hdr.msg_len - len;
 	if (pe_entry->done_len == len && !pe_entry->pe.rx.rx_entry) {
 		fastlock_acquire(&rx_ctx->lock);
-		sock_pe_progress_buffered_rx(rx_ctx);
+		sock_pe_progress_buffered_rx(rx_ctx, -1);
```

Comment:
Don't use magic values like -1 here.  Just pass in INT_MAX (or similar), or define an enum value that's more descriptive (e.g. progress_all).